dmutils/views.py

```diff
@@ -0,0 +1,223 @@
+from abc import ABCMeta, abstractmethod
+import csv
+import enum
+from flask import abort, request, Response
+from flask.views import View
+from io import BytesIO
+from odf.style import TextProperties, TableRowProperties, TableColumnProperties, TableCellProperties, FontFace
+import six
+
+from dmutils import csv_generator
+from dmutils import ods
+
+
+@six.add_metaclass(ABCMeta)
```

Comment:
I don't entirely follow why formally making this an ABC here makes any difference. (But then, I started reading PEP 3119, and lost the will to live, so meh.) The pattern I've seen before is just to name the thing `_Whatever_Base` and just throw some NotImplemented exceptions for the abstract methods.

Reply:
(oh I see, abstract methods aren't abstract at all, by my understanding of the term up until now - they are more like 'must override'. Which feels very weird - if you want to provide some helper functions to assist implementation of specific functions, then those could be separate anyway - possibly in another class altogether.)

Reply:
scratch that, those are examples? I find this confusing

Reply:
Are the comments insufficient? I guess I don't find it confusing but obviously I wouldn't since I've been so close to it for so long. There are example implementations so that it's easy to look at the class and know how to implement the subclass and what the expected returns should be.

Reply:
They *are* examples, but they *are* also invokable through `super`, I just assume the intention here is not to.

Reply:
(hmm actually the documentation seems to be contradictory on whether they can or not, whatever...)